### PR TITLE
Fix/wait for at

### PR DESCRIPTION
**Pull request by dworik**
Fixes issue #1938
## Problem
`first`, `last`, and `at()` delegated directly to Flutter finders, which throw immediately for empty results (`StateError`/`RangeError`). That prevented Patrol’s normal waiting behavior in `tap()`/`waitUntilVisible()` from kicking in when a widget appeared later.
## Fix
Replaced those selectors with a Patrol wrapper that returns an empty result instead of throwing when the selected element is not available yet. This lets waitUntilVisible() keep retrying until the widget appears or the regular timeout is reached.
## Test plan
- [x] dart analyze in `packages/patrol_finders` clean
- [x] flutter test in `packages/patrol_finders/test` - 127/127 passed, including 7 new tests covering: 
   - negative numbers
   - widgets appearing after delay
   - non-existing widgets
- [x] patrol test in `dev/e2e_app/patrol_test/at_finder_test.dart` - successfully waits for a second widget to appear without throwing.

https://github.com/user-attachments/assets/c8b766e7-5968-45d7-84c0-0535f9b08134

## CI
- `test android emulator webview...` expected to fail

### CODE REVIEW

File: packages/patrol_finders/lib/src/custom_finders/patrol_finder.dart [hunk 1]
```diff
@@ -554,7 +554,10 @@ class PatrolFinder implements MatchFinder {
   @override
   PatrolFinder at(int index) {
     // TODO: Throw a better error (https://github.com/leancodepl/patrol/issues/548)
```
Comment (pdenert) [inline packages/patrol_finders/lib/src/custom_finders/patrol_finder.dart]: You can remove this comment. The issue has been closed

File: packages/patrol_finders/lib/src/custom_finders/patrol_finder.dart
Comment (pdenert) [inline packages/patrol_finders/lib/src/custom_finders/patrol_finder.dart:1]: Can we apply same logic for first and last?

File: packages/patrol_finders/test/patrol_finder_test.dart
Comment (pdenert) [inline packages/patrol_finders/test/patrol_finder_test.dart:1]: Please add test with negative index, to check if that behave correctly

**Review by pdenert — COMMENTED** [review]
Please also add video of the demo run of this tests

File: dev/e2e_app/patrol_test/at_finder_test.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      key: _AtFinderItemKey(index),
+      title: Text('Item $index'),
+      onTap: onTap,
+    );
+  }
+}
+
+class _AtFinderItemKey extends ValueKey<int> {
+  const _AtFinderItemKey(super.value);
+}
```
Comment (pdenert) [inline dev/e2e_app/patrol_test/at_finder_test.dart]: This shouldn't be added in test, but directly in the e2e_app. Or you can find and reuse some screen that already exist

File: dev/e2e_app/pubspec.yaml [hunk 3]
```diff
@@ -29,6 +29,8 @@ dependencies:
 
 # Due to: https://github.com/Baseflow/flutter-permission-handler/issues/1450#issuecomment-2673836556
 dependency_overrides:
+  patrol_finders:
+    path: ../../packages/patrol_finders
```
Comment (pdenert) [inline dev/e2e_app/pubspec.yaml]: Revert that. We want to test there real packages

**Review by pdenert — APPROVED** [review]
